tools/pubsys/src/aws/validate_ssm/mod.rs

```diff
@@ -0,0 +1,757 @@
+    let mut parameter_map: HashMap<Region, HashMap<SsmKey, SsmValue>> = HashMap::new();
+    for parameter_list_path in parameter_lists {
+        // Parse the JSON list as a HashMap of region_name, mapped to a HashMap of ami_id, mapped to
+        // a HashMap of parameter_name and parameter_value
+        let parameter_list: HashMap<
+            RegionName,
+            HashMap<AmiId, HashMap<ParameterName, ParameterValue>>,
+        > = serde_json::from_reader(&File::open(parameter_list_path.clone()).context(
+            error::ReadExpectedParameterListSnafu {
+                path: parameter_list_path,
+            },
+        )?)
+        .context(error::ParseExpectedParameterListSnafu)?;
+
+        // Iterate over the parsed HashMap, converting the nested HashMap into a HashMap of Region
+        // mapped to a HashMap of SsmKey, SsmValue
+        parameter_list
+            .iter()
+            .filter(|(region, _)| validation_regions.contains(region))
+            .flat_map(|(region, ami_ids)| {
+                ami_ids
+                    .iter()
+                    .map(move |(ami_id, param_names)| (region, ami_id, param_names))
+            })
+            .flat_map(|(region, ami_id, params)| {
+                params.iter().map(move |(parameter_name, parameter_value)| {
+                    (
+                        region.clone(),
+                        ami_id.clone(),
+                        parameter_name.clone(),
+                        parameter_value.clone(),
+                    )
+                })
+            })
+            .for_each(|(region, ami_id, parameter_name, parameter_value)| {
+                parameter_map
+                    .entry(Region::new(region.clone()))
+                    .or_insert(HashMap::new())
+                    .insert(
+                        SsmKey::new(Region::new(region), parameter_name),
+                        SsmValue {
+                            value: parameter_value,
+                            ami_id,
+                        },
+                    );
+            });
+    }
+    Ok(parameter_map)
+}
+
+/// Common entrypoint from main()
+pub(crate) async fn run(args: &Args, validate_ssm_args: &ValidateSsmArgs) -> Result<()> {
+    let results = validate(args, validate_ssm_args).await?;
+
+    if validate_ssm_args.json {
+        println!(
+            "{}",
+            serde_json::to_string_pretty(&results.get_json_summary())
+                .context(error::SerializeResultsSummarySnafu)?
+        )
+    } else {
+        println!("{}", results)
+    }
+    Ok(())
+}
+
+mod error {
+    use crate::aws::ssm::ssm;
+    use snafu::Snafu;
+    use std::path::PathBuf;
+
+    #[derive(Debug, Snafu)]
+    #[snafu(visibility(pub(super)))]
+    pub enum Error {
+        #[snafu(display("Error reading config: {}", source))]
+        Config { source: pubsys_config::Error },
+
+        #[snafu(display("Error reading validation config at path {:?}: {}", path, source))]
```

Comment:
Paths are weird because they aren't guaranteed to be UTF-8 encoded, here's how you do it:

```suggestion
        #[snafu(display("Error reading validation config at path {}: {}", path.display(), source))]
```